snow/consensus/snowman/mixed_test.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package snowman
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gonum.org/v1/gonum/mathext/prng"
+
+	"github.com/ava-labs/avalanchego/snow/consensus/snowball"
+)
+
+func TestConvergenceSnowFlakeSnowBall(t *testing.T) {
+	require := require.New(t)
+
+	params := snowball.Parameters{
+		K:                     20,
+		AlphaPreference:       11,
+		AlphaConfidence:       11,
+		Beta:                  20,
+		ConcurrentRepolls:     1,
+		OptimalProcessing:     1,
+		MaxOutstandingItems:   1,
+		MaxItemProcessingTime: 1,
+	}
+
+	for peerCount := 20; peerCount < 2000; peerCount *= 10 {
+		numNodes := peerCount
+
+		t.Run(fmt.Sprintf("%d nodes", numNodes), func(t *testing.T) {
+			n := NewNetwork(params, 10, prng.NewMT19937())
+			for i := 0; i < numNodes; i++ {
+				var sm Consensus
+				if i%2 == 0 {
+					factory := TopologicalFactory{factory: snowball.SnowflakeFactory}
+					sm = factory.New()
+				} else {
+					factory := TopologicalFactory{factory: snowball.SnowballFactory}
+					sm = factory.New()
+				}
+
+				require.NoError(n.AddNode(t, sm))
+			}
+
+			for !n.Finalized() {
+				require.NoError(n.Round())
+			}
+
+			require.True(n.Agreement())
```

Comment:
I think it is sufficient to check at this package level, as just snowman alone is pretty high level and complex, and encapsulates in it quite a lot of things, such as transitive voting based on topological sort of the block graph, and this is essentially what is tested here.

I think the only thing not tested here, is what happens if nodes are disconnected or unreachable, but I don't think this should have any difference, because snowball doesn't handle this aspect, but delegates it to snowflake.